sklearn/decomposition/pca.py

```diff
@@ -216,6 +217,77 @@ def __init__(self, n_components=None, copy=True, whiten=False):
+
+        Parameters
+        ----------
+        input_features : list of string, shape (n_features), optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        full_output : boolean or integer, default False
+            When it is "True", return the principal components as the
+            combination of the input features. If "False", will be just
+            the component names. If it is an integer n, it returns the
+            sorted top n contributions to each component.
+
+
+        Returns
+        -------
+        output_feature_names : list of string, shape (n_components)
+            When full_output is "True", it is represented by the contribution
+            of input features and full_output is "False", it just represents
+            the component names
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from sklearn.decomposition import PCA
+        >>> X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
+        >>> pca = PCA(n_components=2).fit(X)
+        >>> pca.get_feature_names(full_output=True)
+        ['0.84*x0 + 0.54*x1', '0.54*x0 - 0.84*x1']
+        >>> pca.get_feature_names(full_output=1)
+        ['0.84*x0', '-0.84*x1']
+        >>> pca.get_feature_names()
+        ['pc0', 'pc1']
+        """
+        check_is_fitted(self, 'components_')
+
+        n_features = self.components_.shape[1]
+        components = self.components_
+        contribution = np.argsort(-np.abs(components), axis=1)
+
+        if input_features is None:
+            input_features = ['x%d' % i for i in range(n_features)]
+        else:
+            if len(input_features) != n_features:
+                raise ValueError("Length of input_features is {0} but it must"
+                    "equal number of features when fitted: {1}.".format
+                    (len(input_features), self.n_features))
+
+        def name_generator(coefficients, names):
+            yield "{0:.2g}*{1}".format(coefficients[0], names[0])
+            for c, n in zip(coefficients[1:], names[1:]):
+                yield "{0:s} {1:.2g}*{2}".format('-' if c < 0 else '+', abs(c), n)
+
+        if full_output is True:
+            feature_names = [' '.join(name_generator(components[i],input_features))
+                            for i in range(self.n_components)]
+        elif full_output is False:
+            feature_names = ['pc{0}'.format(i) for i in range(self.n_components)]
+        elif isinstance(full_output, six.integer_types) and full_output < n_features:
+            required = contribution[:,:full_output]
+            input_features = np.asarray(input_features)
+            feature_names = [' '.join(name_generator(components[i][required[i]], input_features[required[i]]))
+                            for i in range(self.n_components)]
+        else:
+            raise ValueError("full_output must be integer or boolean")
```

Comment:
Oops my bad. Would it be okay to check for this condition and raise the error in the previous elif block ?